[Project] Add ACCV workshop 1st Solution (#1245)
open-mmlab/mmclassification: projects/fgia_accv2022_1st/README.md

# Solution of FGIA ACCV 2022(1st Place)

> This is the 1st Place Solution for Webly-supervised Fine-grained Recognition, refer to the ACCV workshop competition in https://www.cvmart.net/race/10412/base.

Reviewer: Better indicate this is the fine-tuning solution. For pre-training, you can put a link pointing to MMSelfSup.

Author: done

## Result

<details>

<summary>Show the result</summary>

<br>

**LB A**

![LB-A](https://user-images.githubusercontent.com/18586273/205498131-5728e470-b4f6-43b7-82a5-5f8e3bd5168e.png)

**LB B**

![LB-B](https://user-images.githubusercontent.com/18586273/205498171-5a3a3055-370a-4a8b-9779-b686254ebc94.png)

</br>

</details>

## Reproduce / 复现

> For detailed self-supervised pretrain code, please refer to [MMselfSup](https://github.com/open-mmlab/mmselfsup/tree/dev-1.x/projects/fgia_accv2022_1st).

Reviewer: ```suggestion
For detailed self-supervised pretrain code, please refer to [MMSelfSup](https://github.com/open-mmlab/mmselfsup/tree/dev-1.x/projects/fgia_accv2022_1st).
```

For detailed finetuning and inference code, refer to [this repo](https://github.com/Ezra-Yu/ACCV2022_FGIA_1st).

## Description

> ### Overview Flow

Reviewer: ```suggestion
### Overview of Our Solution
```

![image](https://user-images.githubusercontent.com/18586273/205498371-31dbc1f4-5814-44bc-904a-f0d32515c7dd.png)

> ### Model Select

Reviewer: ```suggestion
### Our Model
```

> - ViT(MAE-pt)   # Pretrained from [**MMSelfSup**](https://github.com/open-mmlab/mmselfsup/tree/dev-1.x/projects/fgia_accv2022_1st).

Reviewer: ```suggestion
- ViT(MAE-pre-train)   # Pretrained from [**MMSelfSup**](https://github.com/open-mmlab/mmselfsup/tree/dev-1.x/projects/fgia_accv2022_1st).
```

> - Swin(21kpt)   # From [MMCls-swin_transformer_v2](https://github.com/open-mmlab/mmclassification/tree/dev-1.x/configs/swin_transformer_v2).

Reviewer: ```suggestion
- Swin-v2(SimMIM-pre-train)   # From [MMCls-swin_transformer_v2](https://github.com/open-mmlab/mmclassification/tree/dev-1.x/configs/swin_transformer_v2).
```

> **Main Arch**

Reviewer: ```suggestion
**The architectures we use **
```

- [SoftMaxEQL](https://paperswithcode.com/paper/the-equalization-losses-gradient-driven)   |   [Code](https://github.com/Ezra-Yu/ACCV_workshop/blob/master/src/models/eql.py)
- FlipTTA [Code](https://github.com/Ezra-Yu/ACCV_workshop/blob/master/src/models/tta_classifier.py)
- clean dataset
- self-emsemble: [Uniform-model-soup](https://arxiv.org/abs/2203.05482) | [code](https://github.com/Ezra-Yu/ACCV_workshop/blob/master/tools/model_soup.py)
- [pseudo](https://lilianweng.github.io/posts/2021-12-05-semi-supervised/)  | [Code](https://github.com/Ezra-Yu/ACCV_workshop/blob/master/tools/creat_pseudo.py)
- bagging-emsemble [Code](https://github.com/Ezra-Yu/ACCV_workshop/blob/master/tools/emsemble.py),
- post-process: [re-distribute-label](https://github.com/Ezra-Yu/ACCV_workshop/blob/master/tools/re-distribute-label.py);

![Overview](https://user-images.githubusercontent.com/18586273/205498258-e5720d83-7006-4aea-86b5-aab1a8998c6c.png)

![image](https://user-images.githubusercontent.com/18586273/205498027-def99b0d-a99a-470b-b292-8d5fc83111fc.png)

> #### Used but no improvement

Reviewer: ```suggestion
#### Used but no improvements
```

> 1. Retrieval Paradigm

Reviewer: Cluster-based data filtering?

Author: no, After extracting the features, compare them in the training set and find the most similar samples in the train

2. EfficientNetv2

#### Not used but worth to do

> 1. DiVE for Long tail Problem;

Reviewer: What does ``Dive for`` mean?

> 2. Use SimMIM to train a swinv2 SelfSup pretrined model;

Reviewer: ```suggestion
2. Use SimMIM to pre-train Swin-v2 on the competition dataset
```

> 3. improve re-distribute-label tool

Reviewer: ```suggestion
3. refine the re-distribute-label tool
```
